components/director/internal/domain/eventing/service.go

```diff
@@ -70,6 +80,59 @@ func (s *service) CleanupAfterUnregisteringApplication(ctx context.Context, appI
 	return model.NewEmptyApplicationEventingConfig()
 }
 
+func (s *service) DeleteAfterRemovingRuntimeFromScenarios(ctx context.Context, runtimeID, tenantID string, currentRuntimeScenarios, scenariosToRemove []string) error {
+	appLabels, err := s.scenariosService.GetApplicationScenarioLabelsForAnyMatchingScenario(ctx, scenariosToRemove)
+	if err != nil {
+		return err
+	}
+	for _, label := range appLabels {
+		applicationScenarios, err := labelpkg.ValueToStringsSlice(label.Value)
+		if err != nil {
+			return err
+		}
+		if len(str.IntersectSlice(currentRuntimeScenarios, applicationScenarios)) > 1 { // there is more than one common scenario
+			break
+		} else {
+			defaultEventingLabelKey := GetDefaultEventingForAppLabelKey(uuid.MustParse(label.ObjectID))
+			err = s.labelRepo.Delete(ctx, tenantID, model.RuntimeLabelableObject, runtimeID, defaultEventingLabelKey)
+			if err != nil && !apperrors.IsNotFoundError(err) {
+				return errors.Wrapf(err, "while deleting deefault eventing label")
+			}
```

Comment:
Again when the runtime is removed from scenario it is no longer the default eventing for this application. However, the application can has more runtimes in scenario and one of them should be elected as the defaultEventing when this one is removed.